### PR TITLE
Feature/tis integration

### PR DESCRIPTION
**Pull request by burzum619**
Adding OAuth Rest Template for interacting with TIS.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -25,6 +25,7 @@
         <integration-tests.skip>true</integration-tests.skip>
         <maven.release.plugin.version>2.5.3</maven.release.plugin.version>
         <jaxb-maven.version>2.3.1</jaxb-maven.version>
+        <spring.security.oauth.version>2.1.0.RELEASE</spring.security.oauth.version>
```
Comment (wtritch) [inline pom.xml]: If this is using spring boot, do we need to manually declare this version?  Isn't there an oauth-boot-starter or something?

File: src/main/java/tds/exam/results/configuration/web/security/OAuthSecurityConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
+        requestFactory.setConnectionRequestTimeout(60000);
+        requestFactory.setReadTimeout(60000);
+        return requestFactory;
+    }
+
+    @Bean
+    public OAuth2RestOperations oauthRestTemplate(final OAuth2ClientContext context) {
+        OAuth2RestTemplate template = new OAuth2RestTemplate(resourceDetails(), context);
+        //TODO: Do we need a custom factory? SRP factory has connectionTimeout = 20000 and readTimeout = 60000
```
Comment (wtritch) [inline src/main/java/tds/exam/results/configuration/web/security/OAuthSecurityConfiguration.java]: Recommend removing TODO and annotating the ClientRequestFactory bean method with that information.

File: src/main/java/tds/exam/results/configuration/web/security/OAuthSecurityConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    public ClientHttpRequestFactory clientHttpRequestFactory() {
+        HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
+        requestFactory.setConnectionRequestTimeout(60000);
+        requestFactory.setReadTimeout(60000);
+        return requestFactory;
+    }
+
+    @Bean
+    public OAuth2RestOperations oauthRestTemplate(final OAuth2ClientContext context) {
+        OAuth2RestTemplate template = new OAuth2RestTemplate(resourceDetails(), context);
```
Comment (wtritch) [inline src/main/java/tds/exam/results/configuration/web/security/OAuthSecurityConfiguration.java]: nit, final =)

File: src/main/java/tds/exam/results/repositories/impl/RemoteTestIntegrationSystemRepository.java [hunk 4]
```diff
@@ -19,27 +19,28 @@
...
     @Override
     public void sendResults(final UUID examId, final String results) {
-        if (properties.isSendToTis()) {
+        if (!properties.isSendToTis()) {
             log.info("TIS XML not sent: " + results);
             return;
         }
 
         HttpHeaders headers = new HttpHeaders();
         headers.setContentType(MediaType.APPLICATION_XML);
```
Comment (wtritch) [inline src/main/java/tds/exam/results/repositories/impl/RemoteTestIntegrationSystemRepository.java:40]: Do we need to / should we set the Accept header as well?

File: src/main/java/tds/exam/results/configuration/web/security/OAuthSecurityConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    private String clientId;
+
+    @Value("${oauth.tis.client.secret}")
+    private String clientSecret;
+
+    @Value("${oauth.tis.username}")
+    private String username;
+
+    @Value("${oauth.tis.password}")
+    private String password;
```
Comment (wtritch) [inline src/main/java/tds/exam/results/configuration/web/security/OAuthSecurityConfiguration.java]: Recommend shoving the above properties into a ConfigurationProperties class, and auto-wiring the properties class in a constructor. (And annotate this class with a
`@EnableConfigurationProperties({MyOAuthConfigurationProperties.class})` 
to get it loaded.)

The general rule-of-thumb in Spring 4+ is that @Value is bad.

File: src/main/java/tds/exam/results/configuration/ExamResultsTransmitterServiceProperties.java [hunk 6]
```diff
@@ -68,6 +69,19 @@ public void setTisUrl(final String tisUrl) {
         this.tisUrl = tisUrl;
     }
 
+    /**
+     * Gets the callback url for the Test Integration System
+     *
+     * @return the TIS callback URL
+     */
+    public String getTisCallbackUrl() {
```
Comment (millerts) [inline src/main/java/tds/exam/results/configuration/ExamResultsTransmitterServiceProperties.java:77]: How is this used?  And what value do we expect to put here?  Or is this code we've added and we'll need to implement a endpoint to handle the callback?

Comment (jtreuting) [reply src/main/java/tds/exam/results/configuration/ExamResultsTransmitterServiceProperties.java:77]: This value is `http://tds-web-deployment.sbtds.org/student/tisReply` in our performance environment right now.  When TIS receives or scores a TRT it callsback to TDS to inform it and i think audit records are update.  I'm not sure if it is for scoring or just getting the result in the first place though.

Comment (wtritch) [reply src/main/java/tds/exam/results/configuration/ExamResultsTransmitterServiceProperties.java:77]: Ah shoot, I totally spaced on that.  From an architectural standpoint, as the sender, this service should be handling the callback.  Even if it just forwards the response back to the TDS_ExamService (maybe via additional messaging.)

**Pull request merged**